students/guillaume/quarter_2_project/leankitbackend/leankitbackend/leankitcards.py

```diff
@@ -0,0 +1,297 @@
+        return sorted_projects(self)
+
+    def in_progress(self):
+        return self.sorted_projects()['In Progress']
+
+    @property
+    def date(self):
+        return self._date
+
+    @date.setter
+    def date(self, datetime_obj):
+        if isinstance(datetime_obj, datetime):
+            self._date = datetime_obj
+            # self.year = self.date.year
+            # self.week_n = self.date.isocalendar[1]
+
+    def __repr__(self):
+        if hasattr(self, 'cards_list'):
+            return ('').join([card.__repr__() for card in self.cards_list])
+        return 'There is no cards being imported'
+
+
+class Leankit_Card:
+    def __init__(self, card):
+        if isinstance(card, dict):
+            self.json = card
+            for key, value in card.items():
+                setattr(self, key, value)
+            self.status = True
+        else:
+            self.status = False
+
+    def __repr__(self):
+        try:
+            Name = ''
+            if len(self.assignedUsers) > 0:
+                tmp_lst = [user['fullName'] for user in self.assignedUsers]
+                Name = (', ').join(tmp_lst)
+            tmp_lst = self.type['title'], self.id, self.title, Name
+            return '{} {} {} {}\n'.format(*tmp_lst)
+        except (AttributeError, KeyError):
+            return 'input was not a card'
+
+
+class Leankit_Board:
+    def __init__(self, board):
+        if isinstance(board, dict):
+            self.json = board
+            for key, value in (self.json).items():
+                setattr(self, key, value)
+            self.status = True
+        else:
+            self.status = False
+
+    def __repr__(self):
+        try:
+            tmp_lst = [self.id, self.title]
+            ret_str = ' '.join(tmp_lst) + '\n'
+            ret_str += (('{}\n') * len(self.json)).format(*self.json.keys())
+            return ret_str
+        except AttributeError:
+            return 'Bad Data'
+
+    def users_list(self):
+        try:
+            return [user['fullName'] for user in self.users]
+        except (KeyError, AttributeError):
+            return []
+
+    def lanes_list(self):
+        def f(lane):
+            a = [lane['id'], lane['name'], lane['parentLaneId']]
+            return '{} {} {}'.format(*a)
+
+        try:
+            return [f(lane) for lane in self.lanes]
+        except (KeyError, AttributeError):
+            return []
+
+    def lanes_structures(self):
+
+        def f(key, lst):
+            if key in lst:
+                return lst.index(key)
+
+        def la(lane):
+            return [lane['name'], lane['id'], lane['index']]
+        try:
+            lst_1 = [la(lane) for lane in self.lanes
+                     if lane['parentLaneId'] is None]
+            d = defaultdict(list)
+            for lane in self.lanes:
+                if lane['parentLaneId'] is not None:
+                    d[lane['parentLaneId']].append(la(lane))
+
+            def g(key):
```

Comment:
could some of these functions that unpack data structures be methods of a class instead? that way, only the class needs to know how the data is structured.